src/sagemaker/model.py

```diff
@@ -553,12 +593,20 @@ def __init__(
         self.entry_point = entry_point
         self.source_dir = source_dir
         self.dependencies = dependencies or []
+        self.git_config = git_config
         self.enable_cloudwatch_metrics = enable_cloudwatch_metrics
         self.container_log_level = container_log_level
         if code_location:
             self.bucket, self.key_prefix = fw_utils.parse_s3_url(code_location)
         else:
             self.bucket, self.key_prefix = None, None
+        if self.git_config:
+            updates = git_utils.git_clone_repo(
+                self.git_config, self.entry_point, self.source_dir, self.dependencies
```

Comment:
You should have a none check in the `git_clone_repo` method to at least handle none-value of `entry_point` Do the same thing for `_validate_git_config ` to check `git_config`.